sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c

```diff
@@ -213,6 +333,218 @@ void gnrc_ipv6_nib_rtr_adv_pio_cb(gnrc_netif_t *upstream, const ndp_opt_pi_t *pi
         return;
     }
 
-    _configure_subnets(subnets, upstream, pio);
+#if IS_USED(MODULE_GNRC_IPV6_AUTO_SUBNETS_SIMPLE)
+    /* if we are the only router on this bus, we can directly choose a prefix */
+    _configure_subnets(subnets, 0, upstream, pio);
+#else
+
+    /* store PIO information for later use */
+    _pio_cache = *pio;
+    _upstream  = upstream;
+
+    /* start advertising by sending timeout message to the server thread */
+    msg_t m = {
+        .type = SERVER_MSG_TYPE_TIMEOUT
+    };
+
+    msg_send(&m, _server_pid);
+#endif /* !IS_USED(MODULE_GNRC_IPV6_AUTO_SUBNETS_SIMPLE) */
+}
+
+#if !IS_USED(MODULE_GNRC_IPV6_AUTO_SUBNETS_SIMPLE)
+/**
+ * @brief Check if memory region is set to 0
+ *
+ * @param[in]   The memory array to check
+ * @param[in]   The size of the memory array
+ *
+ * @return  true if all bytes are set to 0
+ */
+static bool _is_empty(const uint8_t *addr, size_t len)
```

Comment:
What `_is_empty`?

```suggestion
static bool _array_is_empty(const uint8_t *addr, size_t len)
```

e.g. would be better.